services/venice-controller/src/main/java/com/linkedin/venice/controller/server/StoresRoutes.java

```diff
@@ -1015,4 +1021,33 @@ public void internalHandle(Request request, ControllerResponse veniceResponse) {
       }
     };
   }
+
+  public Route removeStoreFromGraveyard(Admin admin) {
+    return (request, response) -> {
+      ControllerResponse responseObject = new ControllerResponse();
+      response.type(HttpConstants.JSON);
+      try {
+        if (!isAllowListUser(request)) {
+          response.status(HttpStatus.SC_FORBIDDEN);
+          responseObject.setError(ACL_CHECK_FAILURE_WARN_MESSAGE_PREFIX + request.url());
+          responseObject.setErrorType(ErrorType.BAD_REQUEST);
+          return AdminSparkServer.OBJECT_MAPPER.writeValueAsString(responseObject);
+        }
+        AdminSparkServer.validateParams(request, REMOVE_STORE_FROM_GRAVEYARD.getParams(), admin);
+        String clusterName = request.queryParams(CLUSTER);
+        String storeName = request.queryParams(NAME);
+        responseObject.setCluster(clusterName);
+        responseObject.setName(storeName);
+        admin.removeStoreFromGraveyard(clusterName, storeName);
+      } catch (PreconditionCheckFailedException exception) {
+        responseObject.setError(exception);
+        response.status(exception.getHttpStatusCode());
+        request.attribute(AdminSparkServer.REQUEST_SUCCEED, false);
```

Comment:
`AdminSparkServer.handleError` adds an extra log:
```
LOGGER.error(errMsg, e);
```
Per previous comment in RB3612134, we want to avoid adding many logs into parent controller log files when store graveyard is not ready to be deleted.

Reply:
Can we instead add a flag to `handleError` to denote whether to log errors or not? If not, it is okay to keep it how you have it currently

Reply:
Sure